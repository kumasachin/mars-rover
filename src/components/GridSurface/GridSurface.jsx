import React, { useState, useEffect } from "react";
import { GridRow } from "../../modules";
import "./GridSurface.css";

export const GridSurface = ({
  dimension,
  lostCell = {},
  robotPosition = {},
}) => {
  return (
    <table
      cellSpacing="0"
      border="1"
      width="100%"
      height="100%"
      className="grid-container"
    >
      <tbody>
        <GridRow
          dimension={dimension}
          lostCell={lostCell}
          robotPosition={robotPosition}
        />
      </tbody>
    </table>
  );
};

export default GridSurface;
