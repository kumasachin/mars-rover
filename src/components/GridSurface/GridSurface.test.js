import { render, screen, cleanup } from '@testing-library/react';
import GridSurface from './';

describe("Robots", () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<GridSurface />)
    expect(asFragment(<GridSurface />)).toMatchSnapshot()
  });

  it('should take a snapshot', () => {
    const { asFragment } = render(<GridSurface />)
    expect(asFragment(<GridSurface />)).toMatchSnapshot()
  });
});