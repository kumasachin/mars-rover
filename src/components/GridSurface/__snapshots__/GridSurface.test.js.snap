// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trades should take a snapshot 1`] = `
<DocumentFragment>
  <table
    border="1"
    cellspacing="0"
    class="grid-container"
    height="100%"
    width="100%"
  >
    <tbody />
  </table>
</DocumentFragment>
`;
